### PR TITLE
899 add gamma distribution and other parameters to state_age_function

### LINKED ISSUE #899: Add gamma distribution and other parameters to state_age_function
### Feature description

Add gammasurvival and LogNormDistribution and parameters loc and scale to state_age_function

### Additional context

Code can be copied from #771 
Files:
- cpp/memilio/epidemiology/state_age_function.h
- cpp/tests/test_state_age_function.cpp

### Checklist

- [X] Attached labels, especially loc:: or model:: labels.
- [X] Linked to project.

### PR DESCRIPTION
**Pull request by lenaploetzke**
# Changes and Information

Please **briefly list the changes** (main added features, changed items, or corrected bugs) made:

- added the parameters location and scale to StateAgeFunction 
-  added three state age functions 
- extended the boost archive of memilio (necessary for the state age functions)
- added tests

## Merge Request - Guideline Checklist

Please check our [git workflow](https://github.com/SciCompMod/memilio/wiki/git-workflow). Use the **draft** feature if the Pull Request is not yet ready to review.

### Checks by code author

- [x] Every addressed issue is linked (use the "Closes #ISSUE" keyword below)
- [x] New code adheres to [coding guidelines](https://github.com/SciCompMod/memilio/wiki/Coding-guidelines)
- [ ] No large data files have been added (files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [x] Tests are added for new functionality and a local test run was successful
- [x] Appropriate **documentation** for new functionality has been added (Doxygen in the code and Markdown files if necessary)
- [x] Proper attention to licenses, especially no new third-party software with conflicting license has been added
- [ ] (For ABM development) Checked [benchmark results](https://github.com/SciCompMod/memilio/wiki/Agent-Based-Model-Development) and ran and posted a local test above from before and after development to ensure performance is monitored.

### Checks by code reviewer(s)

- [ ] Corresponding issue(s) is/are linked and addressed
- [ ] Code is clean of development artifacts (no deactivated or commented code lines, no debugging printouts, etc.)
- [ ] Appropriate **unit tests** have been added, CI passes, code coverage and performance is acceptable (did not decrease)
- [ ] No large data files added in the whole history of commits(files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [ ] On merge, add 2-5 lines with the changes (main added features, changed items, or corrected bugs) to the merge-commit-message. This can be taken from the **briefly-list-the-changes** above (best case) or the separate commit messages (worst case).

Closes #899.

### CODE REVIEW

File: cpp/thirdparty/boost_1_75_0.tar.gz
Comment (mknaranja) [inline cpp/thirdparty/boost_1_75_0.tar.gz:1]: I think there is a **big** problem. Did you recommit a changed boost? (+46 MB?)

Comment (lenaploetzke) [reply cpp/thirdparty/boost_1_75_0.tar.gz:1]: Yes i did. Additional functionality of boost is used for the GammaSurvivalFunction.

**Comment by lenaploetzke** [comment]
Attention: this PR should be recreated without the new boost file in the history.

**Comment by mknaranja** [comment]
@lenaploetzke Could you recreate a fresh branch and push all your changes except the boost file to the new branch and also create a new PR?

**Comment by lenaploetzke** [comment]
> @lenaploetzke Could you recreate a fresh branch and push all your changes except the boost file to the new branch and also create a new PR?

Yes, see comment above.

**Comment by lenaploetzke** [comment]
Moved to #1004

**Pull request closed without merging**